.github/workflows/build-and-deploy-staging.yml

```diff
@@ -0,0 +1,146 @@
+name: Deploy Auctions UI staging
+
+on:
+    push:
+      branches: [ main ]
+
+jobs:
+  build-frontend:
+    name: build frontend
+    runs-on: ubuntu-latest
+    permissions: write-all
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Login to GitHub Packages
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ secrets.GH_WORKFLOW_USER }}
+        password: ${{ secrets.GH_WORKFLOW_TOKEN }}
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        aws-region: eu-central-1
+        role-to-assume: arn:aws:iam::802385070966:role/GithubActionsSSMRole
+        role-session-name: AuctionsUiSession
+  
+    - name: Get secrets from parameterstore
+      uses: dkershner6/aws-ssm-getparameters-action@v2
+      with:
+        parameterPairs: "/auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/rpc_url = RPC_URL, /auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/contact_email = CONTACT_EMAIL, /auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/frontend_origin = FRONTEND_ORIGIN, /auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/infura_project_id = INFURA_POJECT_ID, /auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/heapio_id = HEAPIO_ID, /auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/production_domain = PRODUCTION_DOMAIN, /auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/staging_banner_url = STAGING_BANNER_URL"
```

Comment:
This is a new step i added to make sure to securely pull the SSM Secrets and parse them into a Decrypted ENV to use them as `build_args` in the build step.

Reply:
So currently, in order to add new secret one need to:
- Add it to `parameterPairs` here
- Add it to `build-args` [below](https://github.com/sidestream-tech/unified-auctions-ui/pull/607/files#r1698823847)
- Add it to a values file in a [separate repo](https://github.com/sidestream-tech/k8s-projects/blob/main/auctions-ui/staging/values-bot.yml)

Is that it or did I miss any steps?

Can this manual flow be simplified, at least the part with the separate repo? It would be great if by default all secrets from the specified secrets path are provided to the environment with the same name as in the path

Reply:
Ah i can see why this is confusing, let me explain:
**Frontend** REQUIRES the Secrets being baked in during the build process. To pull the Secrets out of the SSM i run the action `aws-ssm-getparameter-action` above to pull the Secrets and export them into an ENV Var (/my/parameter/path = ENV_VAR).
The exported ENV_VAR can now be parsed as a `build_arg` during the Build Step below.

**Bot** doesnt require the Secrets being baked in as build_args but instead can work with the Secrets being provided during runtime. And thats whats being done in the [value-file](https://github.com/sidestream-tech/k8s-projects/blob/39be2a28b75bd8f41762d87f2db77ae32a5aabd8/auctions-ui/staging/values-bot.yml#L62). Unfortunatly that approach doesnt work for the Frontend due to the way its being coded.

> Can this manual flow be simplified, at least the part with the separate repo?

The Integration into ArgoCD kind of requires the GitOps approach of having a `source-of-truth` repo which is `k8s-projects` in our case. What exactly bothers you on this aproach?

> It would be great if by default all secrets from the specified secrets path are provided to the environment with the same name as in the path

IMO its much better to have this amount of granularity to see excatly when which Secret is being used an pulled expecially for people who want to try to understand what is going on exactly.

Reply:
Alright, we can put this simplification requirement out of scope. I just wish the non-trivial and not-testable process of adding new secrets will be documented somewhere very clearly. Could you please add necessary steps for both services (please check my proposed formulations for correctness):

- To the `frontend` readme, under the section ["Environment variables"](https://github.com/sidestream-tech/unified-auctions-ui/blob/6aa81787e72e3f6042e4d19e82e01f7b5ef2a358/frontend/README.md?plain=1#L40)
  ```
  Notes: 
  - Env variables are accessible via the `secret` command under `auction-ui/${environment}/frontend`
  - In order to provide environment variables to the frontend inside the CI, they have to be available at _build time_. For this 1) modify `build-and-deploy-staging.yml` file, `parameterPairs` variable to include comma-separated key-value pairs like `/auction-ui/main.auction-ui.k8s.sidestream.tech/frontend/rpc_url = RPC_URL` 2) modify `build-and-deploy-staging.yml` file, `build-args` variable to include newline-separated key-value pairs like `RPC_URL=${{ env.RPC_URL }}`
  ```
- To the `bot` readme, under the same section ["Environment variables"](https://github.com/sidestream-tech/unified-auctions-ui/blob/6aa81787e72e3f6042e4d19e82e01f7b5ef2a358/bot/README.md?plain=1#L50)
  ```
  Notes:
  - Env variables are accessible via the `secret` command under `auction-ui/${environment}/bot`
  - In order to provide environment variables to the bot inside the CI, they have to be available at _runtime_. For this: 1) modify `https://github.com/sidestream-tech/k8s-projects/blob/main/auctions-ui/staging/values-bot.yml` file, `externalSecrets` variable to include name and path to value like `- name: rpc_url` with `ssmPath: /auction-ui/main.auction-ui.k8s.sidestream.tech/bot/rpc_url` on the next line

Then, I'm fine to merge this one

Reply:
Added you proposed README Notes [here](https://github.com/sidestream-tech/unified-auctions-ui/pull/607/commits/6f13c3015a93440a2707f16919c7901763342842)

Reply:
This is how added notes are [currently rendered](https://github.com/sidestream-tech/unified-auctions-ui/blob/6f13c3015a93440a2707f16919c7901763342842/bot/README.md#environment-variables) (in both files). Please put them _under_ the list with environment variables and make sure they are correctly rendered

<img width="1011" alt="Screenshot 2024-08-01 at 18 26 02" src="https://github.com/user-attachments/assets/b93e2541-14e0-4c6a-9b7b-e5dce19dace9">